deploy/lib/xquery/setup.xqy

```diff
@@ -469,31 +469,36 @@ declare function setup:rewrite-config($import-configs as element(configuration)+
   return setup:suppress-comments($config)
 };
 
-declare function setup:do-setup($import-config as element(configuration)+) as item()*
+declare function setup:do-setup($import-config as element(configuration)+, $options as xs:string) as item()*
 {
+  let $options := if(fn:empty($options) or $options eq "") then ("all") else fn:tokenize($options, ",")
+  let $_ := xdmp:log("OPTIONS: ")
+  let $_ := xdmp:log($options)
+  return
   try
   {
     let $import-config := setup:rewrite-config($import-config)
     return (
-      setup:create-ssl-certificate-templates($import-config),
-      setup:create-privileges($import-config),
-      setup:create-roles($import-config),
-      setup:create-users($import-config),
-      setup:create-external-security($import-config),
-      setup:apply-external-security-settings($import-config),
-      setup:create-mimetypes($import-config),
-      setup:create-groups($import-config),
-      setup:configure-groups($import-config),
-      setup:configure-hosts($import-config),
-      setup:create-forests($import-config),
-      setup:create-databases($import-config),
-      setup:attach-forests($import-config),
-      setup:create-amps($import-config),
-      setup:apply-database-settings($import-config),
-      setup:configure-databases($import-config),
-      setup:create-appservers($import-config),
-      setup:apply-appservers-settings($import-config),
-      setup:create-scheduled-tasks($import-config),
+      if(fn:exists(fn:index-of($options, "all")) or fn:exists(fn:index-of($options, "ssl"))) then setup:create-ssl-certificate-templates($import-config) else (),
```

Comment:
The exists(index-of(..)) makes this very difficult to read. How about a little FLWOR on top that converts $options into a hash-map in which you do one simple if (map:exists($options, 'ssl')) then.. (and make the $options such that if $options was empty, all keys are true by default..)